src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Url/Base64UrlDecoder.cs

```diff
@@ -0,0 +1,850 @@
+        /// Decode the span of UTF-8 encoded text represented as Base64Url into binary data.
+        /// </summary>
+        /// <param name="source">The input span which contains UTF-8 encoded text in Base64Url that needs to be decoded.</param>
+        /// <param name="destination">The output span which contains the result of the operation, i.e. the decoded binary data.</param>
+        /// <returns>The number of bytes written into the <paramref name="destination"/>. This can be used to slice the output for subsequent calls, if necessary.</returns>
+        /// <exception cref="ArgumentException">Thrown when the encoded output cannot fit in the <paramref name="destination"/> provided.</exception>
+        /// <exception cref="FormatException"><paramref name="source"/> contains an invalid Base64Url character,
+        /// more than two padding characters, or a non-white space-character among the padding characters.</exception>
+        /// <remarks>
+        /// As padding is optional the input length not required to be a multiple of 4.
+        /// If the input length is not a multiple of 4 the remainders decoded accordingly:
+        /// Remainder of 3 bytes - decoded into 2 bytes data, decoding succeeds.
+        /// Remainder of 2 bytes - decoded into 1 byte data. decoding succeeds.
+        /// Remainder of 1 byte - is invalid input, causes FormatException.
+        /// </remarks>
+        public static int DecodeFromUtf8(ReadOnlySpan<byte> source, Span<byte> destination)
+        {
+            OperationStatus status = DecodeFromUtf8(source, destination, out _, out int bytesWritten);
+
+            if (OperationStatus.Done == status)
+            {
+                return bytesWritten;
+            }
+
+            if (OperationStatus.DestinationTooSmall == status)
+            {
+                throw new ArgumentException(SR.Argument_DestinationTooShort, nameof(destination));
+            }
+
+            throw new FormatException(SR.Format_BadBase64Char);
+        }
+
+        /// <summary>
+        /// Decode the span of UTF-8 encoded text represented as Base64Url into binary data.
+        /// </summary>
+        /// <param name="source">The input span which contains UTF-8 encoded text in Base64Url that needs to be decoded.</param>
+        /// <param name="destination">The output span which contains the result of the operation, i.e. the decoded binary data.</param>
+        /// <param name="bytesWritten">The number of bytes written into the output span. This can be used to slice the output for subsequent calls, if necessary.</param>
+        /// <returns><see langword="true"/> if bytes decoded successfully, otherwise <see langword="false"/>.</returns>
+        public static bool TryDecodeFromUtf8(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesWritten)
+        {
+            OperationStatus status = DecodeFromUtf8(source, destination, out _, out bytesWritten);
+
+            return status == OperationStatus.Done;
```

Comment:
Generally we only have the Try false state represent one reason for failure... this is representing multiple. I suggest it should only return false if the destination is too short and otherwise throw. @bartonjs?